lib/pages/events_page.dart

```diff
@@ -76,11 +91,10 @@ class _EventsPageState extends State<EventsPage> {
           child: ListView.separated(
             physics: const AlwaysScrollableScrollPhysics(),
             padding: EdgeInsets.fromLTRB(0, 10.0, 0, safePadding.bottom),
-            itemCount:
-                eventsProvider.getEventsByTabIndex(_selectedIndex).length,
+            itemCount: getEventsByTabIndex(_selectedIndex).length,
```

Comment:
Hi, thanks a lot again! @hungneox 

With the last changes;
- When today view is selected list is always empty
- Today view includes events happening on other dates than today

And i think the problem is here
since  itembuilder function below is built after theitemcoult part
they are referring to different lists at a time

would probaly more reliable if we move this to top of the build function like;

```dart
final eventsProvider = Provider.of<Events>(context)
final events = _selectedIndex == 0 ? eventsProvider.events : events.todayEvents
```

then we could just use `events` at itemCount `events.index` also at itembuilder ...` events[index]`

Reply:
Hi, I fixed the bug and move `todayEvents` back to event provider. I think the animation is probably default behavior of flutter listview because it it keep track the scroll position of the listview. But I don't really know. You can see it doesn't always animate in this video. https://imgur.com/a/Uig8tk7

Reply:
I tried to tackle this by making separate list views for each tab. Seems to work alright.
Please see https://github.com/hungneox/event_dot_pizza/pull/1 and tell me what do you think.